shell/browser/api/atom_api_session.cc

```diff
@@ -646,6 +652,42 @@ void Session::Preconnect(const gin_helper::Dictionary& options,
                      url, num_sockets_to_preconnect));
 }
 
+#if BUILDFLAG(ENABLE_BUILTIN_SPELLCHECKER)
+base::Value Session::GetSpellCheckerLanguages() {
+  return browser_context_->prefs()
+      ->Get(spellcheck::prefs::kSpellCheckDictionaries)
+      ->Clone();
+}
+
+void Session::SetSpellCheckerLanguages(
+    gin_helper::ErrorThrower thrower,
+    const std::vector<std::string>& languages) {
+  base::ListValue language_codes;
+  for (const std::string& lang : languages) {
+    std::string code = spellcheck::GetCorrespondingSpellCheckLanguage(lang);
+    if (code.empty()) {
+      thrower.ThrowError("Invalid language code provided: \"" + lang +
+                         "\" is not a valid language code");
+      return;
+    }
+    language_codes.AppendString(code);
+  }
+  browser_context_->prefs()->Set(spellcheck::prefs::kSpellCheckDictionaries,
```

Comment:
This could use a in memory preference store and flush to the file system on app shutdown.

Reply:
It's all tied to the prefs store we construct in AtomBrowserContext which has to be the syncable one to fulfil api requirements for the spellchecker classes.  Not sure how easy it is to make it half in memory, also worth noting that this prefs store is required for extensions to work as well (we just use the same logic)

Reply:
Yup, in memory store can mock syncable, you can check the ones used for `session.setProxy`, we can flush it to file system if needed. But I see that you are editing the pref registry directly here, its fine to leave it as such.